### ISSUE TITLE
NativeModule does not exist

### ISSUE THREAD

**Issue by gurmukhp** [wuxudong/react-native-charts-wrapper#118]
I have done everything that you suggested, but I am still getting errors when trying to preview the app.

There are no build errors but in the console log, I have plenty of errors like so:

```
ExceptionsManager.js:73 Warning: Native component for "RNBarChart" does not exist
reactConsoleErrorHandler @ ExceptionsManager.js:73
printWarning @ warning.js:36
warning @ warning.js:60
requireNativeComponent @ requireNativeComponent.js:52
(anonymous) @ BarChart.js:22
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.js:1
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ chartView.js:4
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ coinDetailView.js:5
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.ios.js:16
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:119
_require @ require.js:110
(anonymous) @ userData.js:49
executeApplicationScript @ debuggerWorker.js:40
(anonymous) @ debuggerWorker.js:65
ExceptionsManager.js:73 Warning: Native component for "RNHorizontalBarChart" does not exist
reactConsoleErrorHandler @ ExceptionsManager.js:73
printWarning @ warning.js:36
warning @ warning.js:60
requireNativeComponent @ requireNativeComponent.js:52
(anonymous) @ HorizontalBarChart.js:22
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.js:2
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ chartView.js:4
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ coinDetailView.js:5
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.ios.js:16
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:119
_require @ require.js:110
(anonymous) @ userData.js:49
executeApplicationScript @ debuggerWorker.js:40
(anonymous) @ debuggerWorker.js:65
ExceptionsManager.js:73 Warning: Native component for "RNBubbleChart" does not exist
reactConsoleErrorHandler @ ExceptionsManager.js:73
printWarning @ warning.js:36
warning @ warning.js:60
requireNativeComponent @ requireNativeComponent.js:52
(anonymous) @ BubbleChart.js:19
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.js:3
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ chartView.js:4
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ coinDetailView.js:5
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.ios.js:16
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:119
_require @ require.js:110
(anonymous) @ userData.js:49
executeApplicationScript @ debuggerWorker.js:40
(anonymous) @ debuggerWorker.js:65
ExceptionsManager.js:73 Warning: Native component for "RNCandleStickChart" does not exist
reactConsoleErrorHandler @ ExceptionsManager.js:73
printWarning @ warning.js:36
warning @ warning.js:60
requireNativeComponent @ requireNativeComponent.js:52
(anonymous) @ CandleStickChart.js:19
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.js:4
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ chartView.js:4
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ coinDetailView.js:5
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.ios.js:16
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:119
_require @ require.js:110
(anonymous) @ userData.js:49
executeApplicationScript @ debuggerWorker.js:40
(anonymous) @ debuggerWorker.js:65
ExceptionsManager.js:73 Warning: Native component for "RNLineChart" does not exist
reactConsoleErrorHandler @ ExceptionsManager.js:73
printWarning @ warning.js:36
warning @ warning.js:60
requireNativeComponent @ requireNativeComponent.js:52
(anonymous) @ LineChart.js:19
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.js:5
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ chartView.js:4
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ coinDetailView.js:5
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.ios.js:16
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:119
_require @ require.js:110
(anonymous) @ userData.js:49
executeApplicationScript @ debuggerWorker.js:40
(anonymous) @ debuggerWorker.js:65
ExceptionsManager.js:73 Warning: Native component for "RNPieChart" does not exist
reactConsoleErrorHandler @ ExceptionsManager.js:73
printWarning @ warning.js:36
warning @ warning.js:60
requireNativeComponent @ requireNativeComponent.js:52
(anonymous) @ PieChart.js:34
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.js:6
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ chartView.js:4
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ coinDetailView.js:5
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.ios.js:16
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:119
_require @ require.js:110
(anonymous) @ userData.js:49
executeApplicationScript @ debuggerWorker.js:40
(anonymous) @ debuggerWorker.js:65
ExceptionsManager.js:73 Warning: Native component for "RNRadarChart" does not exist
reactConsoleErrorHandler @ ExceptionsManager.js:73
printWarning @ warning.js:36
warning @ warning.js:60
requireNativeComponent @ requireNativeComponent.js:52
(anonymous) @ RadarChart.js:25
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.js:7
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ chartView.js:4
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ coinDetailView.js:5
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.ios.js:16
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:119
_require @ require.js:110
(anonymous) @ userData.js:49
executeApplicationScript @ debuggerWorker.js:40
(anonymous) @ debuggerWorker.js:65
ExceptionsManager.js:73 Warning: Native component for "RNScatterChart" does not exist
reactConsoleErrorHandler @ ExceptionsManager.js:73
printWarning @ warning.js:36
warning @ warning.js:60
requireNativeComponent @ requireNativeComponent.js:52
(anonymous) @ ScatterChart.js:19
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.js:8
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ chartView.js:4
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ coinDetailView.js:5
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:126
_require @ require.js:110
(anonymous) @ index.ios.js:16
loadModuleImplementation @ require.js:174
guardedLoadModule @ require.js:119
_require @ require.js:110
(anonymous) @ userData.js:49
executeApplicationScript @ debuggerWorker.js:40
(anonymous) @ debuggerWorker.js:65
ExceptionsManager.js:73 Warning: Native component for "RNCombinedChart" does not exist
```

**Comment by rodrigoxavier** [comment]
This error is in Android or IOS?
To configure on Android just follow the [steps in documentation](https://www.evernote.com/shard/s304/sh/5d501d94-a8e0-4309-9866-e2026356a29d/130a7aa8b84d73a6). To configure in IOS i've followed this [step by step guide](https://www.evernote.com/shard/s304/sh/5d501d94-a8e0-4309-9866-e2026356a29d/130a7aa8b84d73a6).

Hope can help you.

**Comment by gurmukhp** [comment]
@rodrigoxavier This is amazing documentation, much better than the ReadMe, please could you do a pull request to update it for everyone to see? I will look at this later today.

Appreciate you taking the time to do this.

**Comment by rodrigoxavier** [comment]
@gurmukhp this documentations is not mine. I found looking for the internet. Now I don't know where. But I think also that is so much better then the official.

**Comment by gurmukhp** [comment]
It worked! Thank you

@wuxudong If I make a pull request are you available to pull it in or is this repo not maintained?

**Comment by kf729ik** [comment]
Hi,
are you guys using Swift 3.2 or 4.0? Because I cannot able to make it work on both. I followed the guide linked by @rodrigoxavier on IOS. 
I have error messages like:
Type 'NSAttributedStringKey' (aka 'NSString') has no member 'font'
Cannot convert value of type '[String : AnyObject]' to expected argument type '[NSAttributedStringKey : Any]?'

Any solution ? Thanks a lot!

**Comment by wuxudong** [comment]
@gurmukhp A detailed integration guide is appreciated.

**Comment by wuxudong** [comment]
I updated README, add a link to the guide. Thanks guys.

**Comment by platonish** [comment]
I was stuck at the error mentioned by @kf729ik, I resolved it by using swift version 4.1 (the available options were 3.3 and 4.1, and it didn't work on 3.3).
In Xcode, select the root of your project, Build Settings tab -> search 'swift language' -> change the Swift Language version
